### PR TITLE
feat-gateway

### PR DESCRIPTION
**Pull request by rohank07**
closes https://github.com/StatCan/daaas/issues/926

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,29 +21,37 @@ helm-repo-bitnami:
 	helm repo add bitnami https://charts.bitnami.com/bitnami
 
 helm-trino:
-	helm upgrade --install -n trino-system \
+	helm upgrade --create-namespace --install -n trino-system \
 		trino ./trino
 
 helm-namespace-controller:
-	kind load docker-image k8scc01covidacr.azurecr.io/namespace-controller:949b8b02d837fcebd64a135d103e61486185b1f4
+	kind load docker-image statcan/namespace-controller:0.0.1 --name trino
```
Comment (wg102) [inline Makefile:28]: Add an explanation; ex:

> This is the name expected by the chart
> This image needs to exist in the local docker and be tagged by this name to avoid the pull image back-off

File: Makefile [hunk 2]
```diff
@@ -8,11 +8,11 @@ helm-install: helm-deploy helm-all
 
 helm-hive: helm-repo-bitnami
 	helm dependency build ./hive-metastore
-	helm upgrade --install -n hive-system \
+	helm upgrade  --install --create-namespace -n hive-system \
```
Comment (wg102) [inline Makefile:11]: Remove new extra space

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-helm-all:	helm-hive helm-minio helm-trino helm-namespace-controller
+helm-all: helm-hive helm-minio helm-trino helm-namespace-controller
```
Comment (wg102) [inline Makefile:1]: I would rename the helm-all as the one command to rule them all.

Whichever way you do it, just for the idea of running make all to do everything.

File: Makefile [hunk 4]
```diff
@@ -21,29 +21,37 @@ helm-repo-bitnami:
...
-	helm install -n trino-system stable statcan/namespace-controller
+	helm upgrade --install -n trino-system stable statcan/namespace-controller
 
-#export PATH=$PATH:/PATHTOREPO/istio-1.9.0/bin
 helm-install-istio:
+	cd istio-1.7.8/
+	export PATH=$PWD/bin:$PATH
 	istioctl install
 
+cilium-install:
```
Comment (wg102) [inline Makefile:37]: Possibly to add to the make all command

Comment (wg102) [inline Makefile:37]: We may also to add a 'why' such as this is why netpols don't work if missing

File: Makefile [hunk 5]
```diff
@@ -21,29 +21,37 @@ helm-repo-bitnami:
...
 helm-namespace-controller:
-	kind load docker-image k8scc01covidacr.azurecr.io/namespace-controller:949b8b02d837fcebd64a135d103e61486185b1f4
+	kind load docker-image statcan/namespace-controller:0.0.1 --name trino
 	helm repo add statcan https://statcan.github.io/charts
-	helm install -n trino-system stable statcan/namespace-controller
+	helm upgrade --install -n trino-system stable statcan/namespace-controller
 
-#export PATH=$PATH:/PATHTOREPO/istio-1.9.0/bin
 helm-install-istio:
+	cd istio-1.7.8/
```
Comment (wg102) [inline Makefile:33]: Add steps above, or in a readme, or something the like

File: ingress-gateway.yaml [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - httpbin.example.com
+    - trino.example.com
+    tls:
+      httpsRedirect: true
+# ---
```
Comment (wg102) [inline ingress-gateway.yaml:28]: Add comment to why this is and why we use it

File: istio-injection.sh [hunk 7]
```diff
@@ -0,0 +1,52 @@
+# kubectl apply -f samples/addons -n istio-system
```
Comment (wg102) [inline istio-injection.sh:1]: Same as the other. Comment :)

Possibly as we discussed, this not being a `.sh` but a 'txt' or 'readme'

File: istio-injection.sh [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+      - 172.21.255.200-172.21.255.250
+EOF
+
+
+curl -L --remote-name-all https://github.com/cilium/cilium-cli/releases/latest/download/cilium-linux-amd64.tar.gz{,.sha256sum}
+sha256sum --check cilium-linux-amd64.tar.gz.sha256sum
+sudo tar xzvfC cilium-linux-amd64.tar.gz /usr/local/bin
+rm cilium-linux-amd64.tar.gz{,.sha256sum}
+
+cilium install
```
Comment (wg102) [inline istio-injection.sh:52]: Already part of the makefile no?

File: istio-injection.sh [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+  namespace: metallb-system
+  name: config
+data:
+  config: |
+    address-pools:
+    - name: default
+      protocol: layer2
+      addresses:
+      - 172.21.255.200-172.21.255.250
+EOF
```
Comment (wg102) [inline istio-injection.sh:44]: should this be a file, and apply done in makefile?

File: trino/templates/configmap-coordinator.yaml [hunk 10]
```diff
@@ -57,7 +57,11 @@ data:
     http-server.https.port={{ .Values.server.config.https.port }}
     http-server.https.keystore.path={{ .Values.server.config.https.keystore.path }}
   {{- end }}
-  {{ .Values.server.coordinatorExtraConfig | indent 4 }}
+    # http-server.process-forwarded=true
+    # http-server.authentication.type=JWT
+    # http-server.authentication.jwt.key-file=http://localhost:8090/openid/v1/jwks
+    # http-server.authentication.jwt.user-mapping.pattern=^system:serviceaccount:([^:]+):default$
+  # {{ .Values.server.coordinatorExtraConfig | indent 4 }}
```
Comment (wg102) [inline trino/templates/configmap-coordinator.yaml:64]: A little comment on how it's to prep for next task

**Review by wg102 — CHANGES_REQUESTED** [review]
TLDR: I suggest adding multiple comments
